core/src/main/scala/org/bitcoins/core/gcs/BlockFilter.scala

```diff
@@ -0,0 +1,111 @@
+package org.bitcoins.core.gcs
+
+import org.bitcoins.core.crypto.DoubleSha256Digest
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.blockchain.Block
+import org.bitcoins.core.protocol.script.{EmptyScriptPubKey, ScriptPubKey}
+import org.bitcoins.core.protocol.transaction.{Transaction, TransactionOutput}
+import org.bitcoins.core.script.control.OP_RETURN
+import org.bitcoins.core.util.BitcoinSUtil
+import scodec.bits.ByteVector
+
+object BlockFilter {
```

Comment:
Shouldn't we be `extend Factory[BlockFilter]`? 

Isn't there a way to read a `BlockFilter` from strictly binary?

Reply:
`object BlockFilter` is the only `BlockFilter`, I'm not sure if we need a factory for it.